integrations/google-analytics-4/lib/index.js

```diff
@@ -0,0 +1,310 @@
+  if (opts.cookieExpiration) {
+    for (var i = 0; i < measurementIds.length; i++) {
+      config.push([
+        'config',
+        measurementIds[i],
+        {
+          cookie_expires: opts.cookieExpiration
+        }
+      ]);
+    }
+  }
+
+  // Cookie Flags
+  // https://developers.google.com/analytics/devguides/collection/ga4/cookies-user-id#cookie_flags
+  if (opts.cookieFlags) {
+    config.push(['set', { cookie_flags: opts.cookieFlags }]);
+  }
+
+  /**
+   * Privacy and Advertising Settings
+   */
+
+  // Disable All Advertising
+  // https://developers.google.com/analytics/devguides/collection/ga4/display-features#disable_all_advertising_features
+  if (opts.disableAllAdvertisingFeatures) {
+    config.push(['set', 'allow_google_signals', false]);
+  }
+
+  // Disable Advertising Personalization
+  // https://developers.google.com/analytics/devguides/collection/ga4/display-features#disable_advertising_personalization
+  if (opts.disableAdvertisingPersonalization) {
+    config.push(['set', 'allow_ad_personalization_signals', false]);
+  }
+
+  // Load gtag.js using the first measurement ID, then configure
+  // using the `config` commands built above.
+  var self = this;
+  this.load({ measurementId: measurementIds[0] }, function() {
+    for (var i = 0; i < config.length; i++) {
+      window.gtag.apply(null, config[i]);
+    }
+
+    self.ready();
+  });
+};
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+GA4.prototype.loaded = function() {
+  return !!(
+    window.ga4DataLayer && Array.prototype.push !== window.ga4DataLayer.push
+  );
+};
+
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Facade.Identify} event
+ */
+GA4.prototype.identify = function(identify) {
+  var opts = this.options;
+  var userPropertyMappings = opts.userProperties;
+
+  var userProperties = {};
+
+  // Map all customer-defined user property mappings.
+  for (var eventField in userPropertyMappings) {
+    if (!userPropertyMappings.hasOwnProperty(eventField)) {
+      continue;
+    }
+
+    var userProp = userPropertyMappings[eventField];
+    var value = identify.proxy(eventField);
+
+    userProperties[userProp] = value;
+  }
+
+    /**
+     * Map the user_id property the Send User ID setting is enabled. Note that the user ID
+     * can be appended as part of the user_properties object instead of being configured by
+     * an explicit command.
+     * https://developers.google.com/analytics/devguides/collection/ga4/cookies-user-id#set_user_id
+     */
+  var userId = identify.userId();
+  if (opts.sendUserId && userId) {
+    userProperties.user_id = userId;
+  }
+
+  if (Object.keys(userProperties).length) {
+    window.gtag('set', 'user_properties', userProperties);
+  }
+};
+
+/**
+ * Group
+ *
+ * @api public
+ * @param {Facade.Group} group
+ */
+GA4.prototype.group = function(group) {
+  var props = reject({
+    group_id: group.groupId()
+  });
```

Comment:
if this can be empty from the reject if there is no group_id() should we not log this event?

Reply:
I think `groupId` is always required for `group` events, [seen here](https://github.com/segmentio/loosely-validate-event/blob/8b247ad20f817645d569d4e664ff372dcc3be129/index.js#L51), so we probably don't need to wrap the props object with `reject`.

Reply:
SG. +1 to removing reject